src/core/container_func.hpp

```diff
@@ -46,4 +46,19 @@ int find_index(Container const &container, typename Container::const_reference i
 	return -1;
 }
 
+/**
+ * Move elements between first and last to a new position, rotating elements in between as necessary.
+ * @param first Iterator to first element to move.
+ * @param last Iterator to (end-of) last element to move.
+ * @param position Iterator to where range should be moved to.
+ * @returns Iterators to first and last after being moved.
+ */
+template <typename TIter>
+auto Slide(TIter first, TIter last, TIter position) -> std::pair<TIter, TIter>
```

Comment:
Is that kinda `std::shift_left, std::shift_right`, except with an interator instead of a distance?

Reply:
Looks like it. In this case it would require iterator-arithmetic.

Reply:
Is the point of this comment that distance might be considered as parameter instead of iterator? That seems to be more work, both in calculating the distance and in mapping the distance back reliably (if it were allowed to be negative, handling overflows, etc).

Or is the point that `std::rotate` can be replaced by `std::shift_left/std::right`? As I do not really see how shift and rotate are similar, as shift 'throws away' the bits shifted out where rotate does not.